src/observers/stores/opentelemetry.py

```diff
@@ -0,0 +1,128 @@
+                    flat.extend(flatten_dict(v, f"{prefix}.{k}"))
+            else:
+                if prefix:
+                    flat[(f"{prefix}.{k}")] = v
+                else:
+                    flat[k] = v
+
+
+def get_version():
+    try:
+        return version("observers")
+    except PackageNotFoundError:
+        return "unknown"
+
+
+@dataclass
+class OpenTelemetryStore(Store):
+    """
+    OpenTelemetry Store
+    """
+
+    # These are here largely to ease future refactors/conform to
+    # the style of the other stores. They have defaults set in the constructor,
+    # but, set here as well.
+    tracer: Optional[Tracer] = None
+    root_span: Optional[Span] = None
+    connected: bool = False
+
+    def __post_init__(self):
+        # if we weren't called by the `connect` constructor, we still need to
+        # `connect`, and the example construction uses the ObjectName() syntax
+        if not self.connected:
+            self.tracer, self.connected = OpenTelemetryStore._connect()
+        # if we initialize a span here, then all subsequent 'add's can be
+        # added to a continuous trace
+        with self.tracer.start_as_current_span(f"{OTEL_NAMESPACE}.init") as span:
+            span.set_attribute("connected", True)
+            self.root_span = span
+
+    def add(self, record: Record):
+        """Add a new record to the store"""
+        with trace.use_span(self.root_span):
+            with self.tracer.start_as_current_span(f"{OTEL_NAMESPACE}.add") as span:
+                # Split out to be easily edited if the record api changes
+                event_fields = [
+                    "assistant_message",
+                    "completion_tokens",
+                    "total_tokens",
+                    "prompt_tokens",
+                    "finish_reason",
+                    "tool_calls",
+                    "function_call",
+                    "tags",
+                    "properties",
+                    "error",
+                    "model",
+                    "timestamp",
+                    "id",
+                ]
+                for field in event_fields:
+                    data = record.__getattribute__(field)
+                    if data:
+                        if type(data) is dict:
+                            intermediate = flatten_dict(data, field)
+                            for k, v in intermediate:
+                                span.set_attribute(k, v)
+                        else:
+                            span.set_attribute(field, data)
+                # Special case for `messages` as it is a list of dicts
+                messages = [str(message) for message in record.messages]
+                span.set_attribute("messages", messages)
+
+    @classmethod
+    def connect(cls):
+        """Alternate constructor for OpenTelemetryStore"""
+        tracer, connected = cls._connect()
+        return cls(tracer, connected)
+
+    @classmethod
+    def _connect(cls):
```

Comment:
can't we just combine this with the `connect` method?

Reply:
Probably! When I started this, there were instantiations both as StoreClass() and as StoreClass.connect(). Because connect wants to create and return an instance, calling it from __post_init__ didn't feel good, but i needed to call the connect work from __post_init__

but i might be misunderstanding something architecturally that would allow me to fix that!

Reply:
Ok I understand better after reading the current version of the argilla store; this work want's to be done exclusively in `__post_init__` to be at parity with argilla. Cando!